src/pip/_internal/download.py

```diff
@@ -839,13 +840,13 @@ def written_chunks(chunks):
         progress_indicator = DownloadProgressProvider(progress_bar,
                                                       max=total_length)
         if total_length:
-            logger.info("Downloading %s (%s)", url, format_size(total_length))
+            logger.info("Downloading %s (%s)", redact_password_from_url(url), format_size(total_length))
```

Comment:
It would be good to call this once above `if show_progress` IMO:

```python
redacted_url = redact_password_from_url(url)
```

It will make the code a bit less cluttered.